core/src/main/java/org/apache/calcite/rel/rules/PruneEmptyRules.java

```diff
@@ -94,7 +94,7 @@ public interface Config extends RelRule.Config {
   public static final RelOptRule UNION_INSTANCE =
       ImmutableUnionEmptyPruneRuleConfig.of()
           .withOperandSupplier(b0 ->
-              b0.operand(LogicalUnion.class).unorderedInputs(b1 ->
+              b0.operand(Union.class).unorderedInputs(b1 ->
```

Comment:
Better leave the default configurations unchanged to avoid accidentally breaking other projects. Projects who wish to enlarge the matching scope can modify the configuration.

Reply:
Only set operator rules configured for Logical operators so it makes a bit inconsistent the default config.
Anyway I got your point and reverted.

Reply:
Hmm, I didn't notice that the other rules were configured for the general operators. Your initial proposal makes sense. Please revert https://github.com/apache/calcite/pull/2915/commits/ad4cfeaf49100f0ce7f5668c0c55fc2fbd8a0a8a and add a note in `history.md` about the slight change in behavior in the default configuration regarding these rules. 

I will merge the PR afterwards; sorry for the back n forth.